2015-10-31  Daniel Patrick O'Laughlin  <olaughli@lnxsrv07.seas.ucla.edu>

        * timespec.h (timespec_cmp) We fixed the overflow problem by storing
	  diff in a signed long long int and by returning positive, negative,
	  or 0 values depending on the situation.

--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec2.h	2015-10-31 00:19:23.590687000 -0700
@@ -45,8 +45,22 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  signed long long int diff = (signed long long int) a.tv_sec - (signed long long int) b.tv_sec;
+  
+  if(a.tv_sec > b.tv_sec)
+    return 1;
+  else if(a.tv_sec < b.tv_sec)
+    return -1;
+  else if(a.tv_sec == b.tv_sec)  //then you can go down to nanosecs
+    {
+      if(a.tv_nsec > b.tv_nsec)
+	return 1;
+      else if(a.tv_nsec < b.tv_nsec)
+	return -1;
+      else  //they are the equal
+	return 0;
+    }
+  
 }
 
 # if ! HAVE_DECL_NANOSLEEP
